core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -1774,12 +1774,31 @@ wasm_instantiate(WASMModule *module, WASMModuleInstance *parent,
         uint8 *memory_data = NULL;
         uint64 memory_size = 0;
         WASMDataSeg *data_seg = module->data_segments[i];
+#if WASM_ENABLE_SHARED_MEMORY != 0
+        bool is_shared_memory;
+#endif
 
 #if WASM_ENABLE_BULK_MEMORY != 0
         if (data_seg->is_passive)
             continue;
 #endif
 
+#if WASM_ENABLE_SHARED_MEMORY != 0
+        /* Currently we have only one memory instance */
+        if (module->import_memory_count > 0) {
+            is_shared_memory =
+                module->import_memories[0].u.memory.flags ? true : false;
```

Comment:
Should we check for the 0x2 flag here as well? Or imported memory cannot have any other flag values other than 0x2?

Reply:
Yes, it is a mistake, thanks a lot for spotting, fixed.